osu.Game/Screens/OnlinePlay/Multiplayer/Participants/ParticipantPanel.cs

```diff
@@ -171,11 +177,7 @@ protected override void OnRoomUpdated()
 
             // If the mods are updated at the end of the frame, the flow container will skip a reflow cycle: https://github.com/ppy/osu-framework/issues/4187
             // This looks particularly jarring here, so re-schedule the update to that start of our frame as a fix.
-            Schedule(() =>
-            {
-                var ruleset = rulesets.GetRuleset(Room.Settings.RulesetID).CreateInstance();
-                userModsDisplay.Current.Value = User.Mods.Select(m => m.ToMod(ruleset)).ToList();
-            });
+            Schedule(() => userModsDisplay.Current.Value = User.Mods.Select(m => m.ToMod(ruleset.CreateInstance())).ToList());
```

Comment:
The previous code created the ruleset once. This code will create it every time. Please revert. I'd hope reverting won't actually change behaviour.

Reply:
fair enough